modules/3d/src/pointcloud/io_obj.cpp

```
@@ -59,35 +72,86 @@ void ObjDecoder::readData(std::vector<Point3f> &points, std::vector<Point3f> &no
             if (splitArr.size() >= 7)
             {
                 Point3f color;
-                if (ss.rdbuf()->in_avail() != 0) {
+                if (ss.rdbuf()->in_avail() != 0)
+                {
                     Point3_<uchar> uc_color;
                     ss >> color.x >> color.y >> color.z;
                     uc_color.x = static_cast<uchar>(std::round(255.f * color.x));
                     uc_color.y = static_cast<uchar>(std::round(255.f * color.y));
                     uc_color.z = static_cast<uchar>(std::round(255.f * color.z));
-                    rgb.push_back(uc_color);
+                    rgbList.push_back(uc_color);
                 }
             }
         }
         else if (key == "vn")
         {
             Point3f normal;
             ss >> normal.x >> normal.y >> normal.z;
-            normals.push_back(normal);
+            nrmList.push_back(normal);
         }
         else if (key == "f")
         {
-            std::vector<int> vertexInd;
+            std::vector<int> vertexInd, normInd, texInd;
             auto tokens = split(s, ' ');
             for (size_t i = 1; i < tokens.size(); i++)
             {
                 auto vertexinfo = split(tokens[i], '/');
-                int index;
-                std::stringstream vs(vertexinfo[0]);
-                vs >> index;
-                vertexInd.push_back(index - 1);
+                std::array<int, 3> idx = { -1, -1, -1 };
+                for (int j = 0; j < (int)vertexinfo.size(); j++)
+                {
+                    try
+                    {
+                        idx[j] = std::stoi(vertexinfo[j]);
+                    }
+                    // std::invalid_exception, std::out_of_range
+                    catch(const std::exception& e)
+                    { }
```

Please do not ignore exceptions. It should be `CV_LOG_ERROR` at least.